src/lib/types/model.ts

```diff
@@ -365,3 +365,66 @@ export interface IFeatureStrategySegment {
     featureStrategyId: string;
     segmentId: number;
 }
+
+export interface ISuggestChangeset {
+    id: number;
+    state: string;
+    project: string;
+    environment: string;
+    createdBy?: Pick<User, 'username' | 'imageUrl'>;
+    createdAt?: Date;
+    changes?: ISuggestChange[];
+    events?: ISuggestChangeEvent[];
+    // approvalFlow: IChangeSetApprovalFlow;
+}
+export interface IApprovalFlow {}
+
+export interface IChangeSetApprovalFlow {
+    id: string;
+    flow: IApprovalFlow;
+}
+
+export interface ISuggestChange {
+    id: number;
+    action: SuggestChangeAction;
+    feature: string;
+    payload?: unknown;
+    createdBy?: Pick<User, 'username' | 'imageUrl'>;
+    createdAt?: Date;
+}
+
+export enum SuggestChangesetEvent {
+    CREATED = 'CREATED',
+    UPDATED = 'UPDATED',
+    SUBMITTED = 'SUBMITTED',
+    APPROVED = 'APPROVED',
+    REJECTED = 'REJECTED',
+    CLOSED = 'CLOSED',
+}
+
+export enum SuggestChangeAction {
+    UPDATE_ENABLED = 'updateEnabled',
+    ADD_STRATEGY = 'strategyAdd',
+    UPDATE_STRATEGY = 'strategyUpdate',
+    DELETE_STRATEGY = 'strategyDelete',
+}
+
+export enum SuggestChangeEvent {
+    UPDATE_ENABLED = 'updateFeatureEnabledEvent',
+    ADD_STRATEGY = 'addStrategyEvent',
+    UPDATE_STRATEGY = 'updateStrategyEvent',
+    DELETE_STRATEGY = 'deleteStrategyEvent',
+}
+
+export interface ISuggestChangeEvent {
+    id: number;
+    event: SuggestChangesetEvent;
+    data: ISuggestChangeEventData;
+    createdBy?: Pick<User, 'username' | 'imageUrl'>;
+    createdAt?: Date;
```

Comment:
This is another instance where these fields are listed as optional. Nothing new to add; just wanted to create a comment as a marker 🤓

Reply:
They were made optional because they are added when commiting to the db - I can create more objects (SuggestChangeCreate and SuggestChangesetCreate and SuggestChangeEventCreate) that do not have them, if you think that this is a problem.

Reply:
Got it. But when it comes out of the DB, it'll always have this data, right? In that case, my vote goes to using more types that are more specific (using TS's fancy type system). That way, you'll always know you have  createdBy/createdAt data when it's coming out of the DB 💁🏼 But I'm willing to be challenged on this.